src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -443,6 +443,13 @@ internal bool EntryIsFresh(ResponseHeaders responseHeaders, TimeSpan age, bool v
 
                 if (EntryIsFresh(cachedResponseHeaders, age, verifyAgainstRequest: true))
                 {
+                    // Check conditional request rules
+                    if (ConditionalRequestSatisfied(cachedResponseHeaders))
+                    {
+                        _httpContext.Response.StatusCode = StatusCodes.Status304NotModified;
+                        return true;
```

Comment:
I had the `responseServed` variable previously because we couldn't short-circuit after failing to serve a response due to the `OnlyIfCached` header. The conditional requests are different, if we served that, we can return right away. But I guess for consistency, I can just keep using `responseServed` and return it at the end. This means one extra check on the boolean even after satisfying the conditional request but I guess that's fine.